### PR TITLE
CIRC-1933: Implementation for Search Slips API

### PR DESCRIPTION
**Pull request by OleksandrVidinieiev**
## Purpose
Create implementation for Search Slips API, which is very similar to the existing Pick Slips API.
[CIRC-1933](https://issues.folio.org/browse/CIRC-1933)

#### TODOS and Open Questions
- [ ] Check logging

### CODE REVIEW

File: src/main/java/org/folio/circulation/resources/SlipsResource.java [hunk 1]
```diff
@@ -68,7 +135,60 @@ protected CompletableFuture<Result<MultipleRecords<Location>>> fetchLocationsFor
...
+      log.info("fetchPagedItemsForLocations:: locationIds is empty");
+
+      return completedFuture(succeeded(emptyList()));
+    }
+
+    List<String> itemStatusValues = itemStatuses.stream()
+      .map(ItemStatus::getValue)
+      .toList();
+
+    Result<CqlQuery> statusQuery = exactMatchAny(STATUS_NAME_KEY, itemStatusValues);
```
Comment (roman-barannyk) [inline src/main/java/org/folio/circulation/resources/SlipsResource.java:161]:
```suggestion
    List<String> itemStatusValues = itemStatuses.stream()
      .map(ItemStatus::getValue)
      .toList();
    Result<CqlQuery> statusQuery = exactMatchAny(STATUS_NAME_KEY, itemStatusValues);
```

File: src/main/java/org/folio/circulation/resources/SlipsResource.java [hunk 2]
```diff
@@ -68,7 +135,60 @@ protected CompletableFuture<Result<MultipleRecords<Location>>> fetchLocationsFor
...
+    ItemRepository itemRepository, LocationRepository locationRepository) {
+
+    log.debug("fetchPagedItemsForLocations:: parameters multipleLocations: {}",
+      () -> multipleRecordsAsString(multipleLocations));
+    Collection<Location> locations = multipleLocations.getRecords();
+
+    Set<String> locationIds = locations.stream()
+      .map(Location::getId)
+      .filter(StringUtils::isNoneBlank)
+      .collect(toSet());
```
Comment (roman-barannyk) [inline src/main/java/org/folio/circulation/resources/SlipsResource.java:149]:
```suggestion
    log.debug("fetchPagedItemsForLocations:: parameters multipleLocations: {}",
      () -> multipleRecordsAsString(multipleLocations));
    Collection<Location> locations = multipleLocations.getRecords();
    Set<String> locationIds = locations.stream()
      .map(Location::getId)
      .filter(StringUtils::isNoneBlank)
      .collect(toSet());
```

File: src/test/java/api/requests/StaffSlipsTests.java [hunk 3]
```diff
@@ -89,51 +99,89 @@ void responseContainsNoPickSlipsForWrongServicePointId() {
...
+      .by(usersFixture.james()));
+
+    Response response = slipsType.get(servicePointId);
+
+    assertThat(response.getStatusCode(), is(HTTP_OK));
+    assertResponseHasItems(response, 0, slipsType);
+  }
+
+  @ParameterizedTest
+  @MethodSource(value = "getAllowedStatusesForHoldRequest")
```
Comment (roman-barannyk) [inline src/test/java/api/requests/StaffSlipsTests.java:142]: what if we just explicitly exclude NONE value from @EnumSource?

Comment (OleksandrVidinieiev) [reply src/test/java/api/requests/StaffSlipsTests.java:142]: And which enum would we use? `ItemStatus` contains many statuses which are not allowed for a Hold request.

**Review by roman-barannyk — APPROVED** [review]

**Pull request merged**